src/censored.jl

```diff
@@ -0,0 +1,168 @@
+"""
+    censored(d::UnivariateDistribution, l::Real, u::Real)
+
+Censor a univariate distribution `d` to the interval `[l, u]`.
+
+The lower bound `l` can be finite or `-Inf` and the upper bound `u` can be finite or
+`Inf`. The function throws an error if `l > u`.
+
+The function falls back to constructing a [`Censored`](@ref) wrapper.
+
+# Implementation
+
+To implement a specialized censored form for distributions of type `D`, the method
+`censored(d::D, l::T, u::T) where {T <: Real}` should be implemented.
+"""
+function censored(d::UnivariateDistribution, l::Real, u::Real)
+    return censored(d, promote(l, u)...)
+end
+
+function censored(d::UnivariateDistribution, l::T, u::T) where {T <: Real}
+    l <= u || error("the lower bound must be less or equal than the upper bound")
+    Censored(d, promote(l, u)...)
+end
+
+"""
+    Censored
+
+Generic wrapper for a censored distribution.
+"""
+struct Censored{D<:UnivariateDistribution, S<:ValueSupport, T <: Real} <: UnivariateDistribution{S}
```

Comment:
As pointed out in https://github.com/JuliaStats/Distributions.jl/issues/1468#issuecomment-1009994772, for a continuous distribution, the censored distribution `censored(dist, l, u)` is equivalent to `MixtureModel([Dirac(l), truncated(dist, l, u), Dirac(u)], [cdf(dist, l), cdf(dist, u) - cdf(dist, l), cdf(dist, u)])`. It might be worth it then to make `Censored` an `AbstractMixtureModel`.

Reply:
I don't think that is needed.